tests/support/kubernetes.py

```diff
@@ -468,6 +522,127 @@ async def delete_namespaced_config_map(
         self._maybe_error("delete_namespaced_config_map", name, namespace)
         return self._delete_object(namespace, "ConfigMap", name)
 
+    # DEPLOYMENT API
+
+    async def create_namespaced_deployment(
+        self, namespace: str, body: V1Deployment
+    ) -> None:
+        """Create a deployment object.
+
+        This will create a ReplicaSet object, which will in turn create
+        a Pod object.
+
+        There is no parsing of ``replicas`` and no attempt to do proper
+        selectors; for purposes of the test, a replicaset and a single pod
+        with the same name as the deployment will be created.
+
+        Parameters
+        ----------
+        namespace
+            Namespace in which to create the deployment.
+        body
+            Deployment specification.
+
+        Raises
+        ------
+        ApiException
+            Raised with 409 status if the deployment already exists.
+        """
+        self._maybe_error("create_namespaced_pod", namespace, body)
+        name = body.metadata.name
+        await self.create_namespaced_replica_set(
+            namespace,
+            V1ReplicaSet(
+                metadata=V1ObjectMeta(name=name, namespace=namespace)
+            ),
+        )
+        body.status = V1DeploymentStatus(available_replicas=1)
+        self._update_metadata(body, "v1", "Deployment", namespace)
+        self._store_object(namespace, "Deployment", name, body)
+
+    async def delete_namespaced_deployment(
+        self, name: str, namespace: str
+    ) -> V1Status:
+        """Delete a deployment object.  This will also delete its
+        corresponding replica_set.
+
+        Parameters
+        ----------
+        name
+            Name of deployment to delete.
+        namespace
+            Namespace of deployment to delete.
+
+        Returns
+        -------
+        V1Status
+            Success status.
+
+        Raises
+        ------
+        ApiException
+            Raised with 404 status if the pod was not found.
+        """
+        await self.delete_namespaced_replica_set(name, namespace)
+        self._maybe_error("delete_namespaced_deployment", name, namespace)
```

Comment:
`_maybe_error` must always be first.

Reply:
All the deployment stuff has gone away since I moved to a Safir branch and I'm not using it and the ReplicaSets anymore.